src/blob/core.rs

```diff
@@ -258,12 +258,13 @@ where
                     _ => e.into(),
                 }
             })?;
-        self.index.push(record.header().clone())?;
+        self.index.push(key, record.header().clone())?;
         self.current_offset += buf.len() as u64;
-        Ok(())
+        Ok(record.header().clone())
```

Comment:
Does the `clone()` needed here? If I understand right: we own `record` so we can return header without cloning